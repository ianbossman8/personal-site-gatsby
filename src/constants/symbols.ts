import THEME from './theme'

const SYMBOLS = {
  [THEME.DARK]: {
    themeSwitch: 0x1f31e // ðŸŒž
  },
  [THEME.LIGHT]: {
    themeSwitch: 0x1f31a // ðŸŒš
  },
  house: 0x1f3e1,
  pointLeft: 0x1f448,
  pointRight: 0x1f449,
  party: 0x1f389,
  writingHand: 0x270d,
  tools: 0x1f528,
  detective: 0x1f575,
  gift: 0x1f381,
  monkey: 0x1f435
}

export default SYMBOLS
