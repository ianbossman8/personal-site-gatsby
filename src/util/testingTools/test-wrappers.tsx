import React, { FC, ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import THEME from '../../constants/theme'
import { styleTheme } from '../../styles/theme'

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={styleTheme[THEME.LIGHT]}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
