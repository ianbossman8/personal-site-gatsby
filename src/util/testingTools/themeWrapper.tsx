import React from 'react'
import { ThemeProvider } from 'styled-components'
import THEME from '../../constants/theme'
import { styleTheme } from '../../styles/theme'

export default function themeWrapper(component: React.ReactElement) {
  return <ThemeProvider theme={styleTheme[THEME.LIGHT]}>{component}</ThemeProvider>
}
