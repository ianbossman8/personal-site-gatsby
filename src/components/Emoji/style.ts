import styled from 'styled-components'
import { SIZE } from '../../constants/font'

export const SymbolSpan = styled.span<{ size: SIZE }>`
  font-size: ${({ theme: { font }, size }) => font.size[size]};
  user-select: none;
`
