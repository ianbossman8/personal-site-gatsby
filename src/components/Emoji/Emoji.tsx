import React from 'react'
import { SymbolSpan } from './style'
import { SIZE } from '../../constants/font'

interface Props {
  label: string
  symbol: number
  size?: SIZE
}

export default function Emoji(props: Props) {
  const { label, symbol, size = SIZE.XL } = props

  return (
    <SymbolSpan role="img" aria-label={label} size={size}>
      {String.fromCodePoint(symbol)}
    </SymbolSpan>
  )
}
