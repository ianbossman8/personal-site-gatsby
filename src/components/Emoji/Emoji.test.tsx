import React from 'react'
import renderer from 'react-test-renderer'
import { screen } from '@testing-library/react'
import 'jest-styled-components'
import Emoji from './Emoji'
import themeWrapper from '../../util/testingTools/themeWrapper'
import { render } from '../../util/testingTools/test-wrappers'
import symbols from '../../constants/symbols'

describe('Emoji', () => {
  describe('snapshot', () => {
    it('should render a house emoji', () => {
      const tree = renderer
        .create(themeWrapper(<Emoji label="testing" symbol={symbols.house} />))
        .toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('render context', () => {
    it('should contain a span', () => {
      render(<Emoji label="testing" symbol={symbols.house} />)

      expect(screen.getByRole('img')).toHaveTextContent(String.fromCodePoint(symbols.house))
    })
  })
})
