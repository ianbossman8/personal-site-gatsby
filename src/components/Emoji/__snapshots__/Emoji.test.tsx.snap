// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji snapshot should render a house emoji 1`] = `
.c0 {
  font-size: 2rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<span
  aria-label="testing"
  className="c0"
  role="img"
  size="XL"
>
  üè°
</span>
`;
